akka-tools-persistence/src/main/scala/no/nextgentel/oss/akkatools/persistence/EnhancedPersistentActor.scala

```diff
@@ -412,10 +504,31 @@ abstract class EnhancedPersistentActor[E:ClassTag, Ex <: Exception : ClassTag]
     try {
       if (doUnconfirmedWarningProcessing && (command.isInstanceOf[AtLeastOnceDelivery.UnconfirmedWarning])) {
         internalProcessUnconfirmedWarning(command.asInstanceOf[AtLeastOnceDelivery.UnconfirmedWarning])
-      } else if (pendingDurableMessage.isDefined && processedDMs.contains( ProcessedDMEvent.createFromDM(pendingDurableMessage.get) )) {
+      } else if (pendingDurableMessage.isDefined && internalDurableState.processedDMs.contains( ProcessedDMEvent.createFromDM(pendingDurableMessage.get) )) {
         onAlreadyProcessedCmdViaDMReceivedAgain(command)
       } else {
-        tryCommand.apply(command)
+        command match {
+          case snapFailure: SaveSnapshotFailure => {
+            onSnapshotFailure(snapFailure)
+          }
+          case snapSuccess: SaveSnapshotSuccess => {
+            deletePotentialMessagesMarkedForDeletion()
+            onSnapshotSuccess(snapSuccess)
+          }
+          case deleteMessagesSuccess: DeleteMessagesSuccess => {
+            //Try to persist the deleted messages mark, and if ok run update to state and inform user

```

Comment:
I think it is better to move this code into the onDeleteMessagesSuccess-method.. or into other method.. Too much logic  placed here..

Reply:
I think I'll add a new method, since onDeleteMessageSuccess is overridable and should not handle EnhancedPersistentActor internals I think.